tgui/packages/tgui/interfaces/Newscaster.js

```diff
@@ -0,0 +1,615 @@
+        <Stack fill vertical>
+          <Stack.Item>
+            <Tabs fluid textAlign="center">
+              <Tabs.Tab
+                color="Green"
+                selected={screenmode === NEWSCASTER_SCREEN}
+                onClick={() => setScreenmode(NEWSCASTER_SCREEN)}>
+                Newscaster
+              </Tabs.Tab>
+              <Tabs.Tab
+                Color="Blue"
+                selected={screenmode === BOUNTYBOARD_SCREEN}
+                onClick={() => setScreenmode(BOUNTYBOARD_SCREEN)}>
+                Bounty Board
+              </Tabs.Tab>
+            </Tabs>
+          </Stack.Item>
+          <Stack.Item grow>
+            {screenmode === NEWSCASTER_SCREEN && (
+              <NewscasterContent />
+            )}
+            {screenmode === BOUNTYBOARD_SCREEN && (
+              <BountyBoardContent />
+            )}
+          </Stack.Item>
+        </Stack>
+      </Window.Content>
+    </Window>
+  );
+};
+
+/** The modal menu that contains the prompts to making new channels. */
+const NewscasterChannelCreation = (props, context) => {
+  const { act, data } = useBackend(context);
+  const [lockedmode, setLockedmode] = useLocalState(context, 'lockedmode', 1);
+  const {
+    creating_channel,
+    name,
+    desc,
+  } = data;
+  if (!creating_channel) {
+    return null;
+  }
+  return (
+    <Modal
+      textAlign="center"
+      mr={1.5}>
+      <Stack vertical>
+        <>
+          <Stack.Item>
+            <Box pb={1}>
+              Enter channel name here:
+              <Button
+                content="X"
+                color="red"
+                position="relative"
+                top="20%"
+                left="15%"
+                onClick={() => act('cancelCreation')} />
+            </Box>
+            <TextArea
+              fluid
+              height="40px"
+              width="240px"
+              backgroundColor="black"
+              textColor="white"
+              maxLength={42}
+              onChange={(e, name) => act('setChannelName', {
+                channeltext: name,
+              })}>
+              Channel Name
+            </TextArea>
+          </Stack.Item>
+          <Stack.Item>
+            <Box pb={1}>
+              Enter channel description here:
+            </Box>
+            <TextArea
+              fluid
+              height="150px"
+              width="240px"
+              backgroundColor="black"
+              textColor="white"
+              maxLength={512}
+              onChange={(e, desc) => act('setChannelDesc', {
+                channeldesc: desc,
+              })}>
+              Channel Description
+            </TextArea>
+          </Stack.Item>
+          <Stack.Item>
+            <Section>
+              Set Channel as Public or Private
+              <Box pt={1}>
+                <Button
+                  selected={!lockedmode}
+                  content="Public"
+                  onClick={() => act(setLockedmode(false))} />
```

Comment:
Act is unnecessary here, you're effectively sending an act with "undefined" string.